geode-management/src/main/java/org/apache/geode/management/configuration/Pdx.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.management.configuration;
+
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+import org.apache.geode.cache.configuration.CacheElement;
+import org.apache.geode.management.api.CorrespondWith;
+import org.apache.geode.management.api.RestfulEndpoint;
+import org.apache.geode.management.runtime.RuntimeInfo;
+
+public class Pdx extends CacheElement implements RestfulEndpoint, CorrespondWith<RuntimeInfo> {
+  public static final String PDX_ID = "PDX";
+  public static final String PDX_ENDPOINT = "/configurations/pdx";
+
+  protected Boolean readSerialized;
```

Comment:
It seems like "private" would be better than "protected" on these fields. Any subclass can use the gettors and settors. Did you have any reason for using "protected" instead of "private"?

Reply:
It's just a personal habit that I make my fields protected so that all subclass don't have to reference them using getter/setters. Seems cleaner to me this way, but I can change it.

Reply:
I tend to favor "private" and prefer even subclasses to use the gettors and settors. This helps if the gettor or settor has additional behaviour and lets you reason about the code when looking at the class (you know that the fields are only accessible by the current class).

But in this case this class will be in the external api and protected stuff is part of the external api.
Another thing to consider with having classes in the external api; if the class is not final then we are allowing users to create subclasses of our external api classes. Do we want to support that or should we make the classes final or move the class to internal and use an interface.